themes/classic/templates/_partials/head.tpl

```diff
@@ -39,7 +39,9 @@
   {if $page.meta.robots !== 'index'}
     <meta name="robots" content="{$page.meta.robots}">
   {/if}
-  {if $page.canonical}
+  {if $page.page_name === 'product'}
```

Comment:
Hi @micka-fdz,

good idea to handle all types of canonical in the head template. However instead of testing the page type it would be cleaner if `$page.canonical` was correctly set even on product page. So it should be handed in the ProductController rather than the template, by adding this method:

```php
public function getCanonicalUrl()
    {
        /** @var \PrestaShop\PrestaShop\Adapter\Presenter\Product\ProductLazyArray $product */
        $product = $this->context->smarty->getTemplateVars('product');

        return $product->getCanonicalUrl();
    }
```
(missing the `use` statement here, to avoid the fully qualified class name, but it's for the example)

Reply:
I agree, it is not the role of the template to define the canonical.

Reply:
I tried to add your method to ProductController and it seems that the `use` is not necessary. What do you think?